docker/docker-tezos-packages.py

```diff
@@ -103,19 +103,23 @@ def validate_dists(dists, versions, target_os):
 if not octez_version:
     raise Exception("Environment variable OCTEZ_VERSION is not set.")
 
-# prebuild docker image before using containers
-check_call(
-    f"""
-{virtualisation_engine}
-build -t tezos-{target_os}
--f docker/package/Dockerfile-{target_os} .
-"""
-)
+# that option reflects launchpad and copr build infrastruture
+# and has a purpose to behave more similar to them
+images = ["36"] if target_os == "fedora" else distributions
```

Comment:
Can you clarify this? I don't get why `36` would be the default value for fedora `images` but `distributions` is picked otherwise

Reply:
As you remember, `copr` build infrastructure uses `mock` for build on `fedora 36`, so it would be better option for our builds.
For ubuntu builds, we build them in impure environments, for now I found it quite complicated to run `pbuilder` under docker. Thus, for better reproducibility, I think we should build `bionic` packages in `bionic` container and etc, because during implementation of this PR, I got some strange errors on `bionic` package installation when build on `focal` container.

Reply:
> copr build infrastructure uses mock for build on fedora 36, so it would be better option for our builds.

Ah, I see, but I'm not sure that we should try to match `copr`, instead I propose that we use the latest available stable fedora. What do you think?

AFAIK one of the features of `mock` is being able to perform a build with some isolation, so it should work on different fedora versions in the same manner.

Also, we aren't sure of what `copr` is using for its builds now or in the future, so I don't this we can reasonably follow what they do.

Maybe picking the lastest available `fedora` version (so `37` now) would be the best option for us

Reply:
Also, it would be nice to have a comment here that's a bit longer and more clear explaining this choice, otherwise we'll likely forget why we did this in the future.

I'd also suggest to mention that as time goes on this fedora version should be updated.

Reply:
> AFAIK one of the features of mock is being able to perform a build with some isolation, so it should work on different fedora versions in the same manner.

We just had the problem with `mock bootstrap` option which showed it's not entirely true even using `mock`. 
On the other hand, docs says that copr uses latest stable fedora, so both ways are about the same.